src/components/SingleArticle.jsx

```diff
@@ -1,30 +1,53 @@
 import { useEffect, useState } from "react";
 import {useParams} from "react-router-dom"
-import { getArticleById } from "../utils/api";
+import { getArticleById, getCommentsById } from "../utils/api";
+import {CommentsCard} from "./CommentsCard"
+import {Votes} from "./CommentsVotes"
 
     const SingleArticle = () => {
      const {article_id} = useParams();
      const [article, setArticle] = useState("");
      const [loading, setLoading] = useState(true);
-
+     const [comments, setComments] = useState("");
+     
     useEffect(()=>{
-        getArticleById(article_id).then((articleFropmApi) => {
+        Promise.all([getArticleById(article_id), getCommentsById(article_id)]).then(([articleFropmApi, commentsFropmApi]) => {
         setLoading(false);
         setArticle(articleFropmApi)
+        setComments(commentsFropmApi)
             
         })
     }, [article_id])
     if (loading) {
         return <h3>Loading...</h3>;
       }
-
+      
     return (
-        <div className="single_article">
-     
+        <div>
+     <div className="single_article">
      <h4> {article.title}  </h4>
     <p>{article.body}</p>
     <img src={article.article_img_url} alt={article.img}></img>
-    
+     <Votes 
+     id={article.article_id}
+     votes={article.votes}/> 
+    </div>
+    <h4 className="comments_header">Comments:</h4>
+      {comments.map((comment) => {
+        return <ul key={comment.comment_id}>
+          <li><CommentsCard 
```

Comment:
You've got the wrong html tags here. The list should be wrapped in the `ul` and each thing inside it should be an `li`. Move the `ul` to outside the map and make the `CommentsCard` return the `li` instead of a `div`